admiral/component_envs/resources.py

```diff
@@ -0,0 +1,110 @@
+
+import numpy as np
+
+from admiral.envs import Agent
+from admiral.component_envs.observer import ObservingAgent
+from admiral.component_envs.component import Component
+
+class GridResourceHarvestingAgent(Agent):
+    def __init__(self, max_harvest=None, **kwargs):
+        assert max_harvest is not None, "max_harvest must be nonnegative number"
+        self.max_harvest = max_harvest
+        super().__init__(**kwargs)
+    
+    @property
+    def configured(self):
+        return super().configured and self.max_harvest is not None
+
+class GridResourceComponent(Component):
+    """
+    Resources exist in the cells of the grid. The grid is populated with resources
+    between the min and max value on some coverage of the region.
+
+    This environment support resource depletion: if a resource falls below the
+    minimum value, it will not regrow.
+    """
+    def __init__(self, region=None, agents=None, coverage=0.75, min_value=0.1, max_value=1.0,
+            regrow_rate=0.04, original_resources=None, **kwargs):        
+        self.original_resources = original_resources
+        if self.original_resources is None:
+            assert type(region) is int, "Region must be an integer."
+            self.region = region
+        else:
+            self.region = self.original_resources.shape[0]
+        self.min_value = min_value
+        self.max_value = max_value
+        self.regrow_rate = regrow_rate
+        self.coverage = coverage
+
+        assert type(agents) is dict, "agents must be a dict"
+        self.agents = agents
+
+        from gym.spaces import Box
+        for agent in self.agents.values():
+            if isinstance(agent, GridResourceHarvestingAgent):
+                agent.action_space['harvest'] = Box(0, agent.max_harvest, (1,), np.float)
+            if isinstance(agent, ObservingAgent):
```

Comment:
What about the case where we want to impose partial observability of the state of the environment by having some agents that cannot observe resources, but can observe other things, such as agent positions. Perhaps framing observations as channel observations will allow us to more easily capture these use cases. Should be considered in #25